### ISSUE TITLE
ImagePullBackOff for pruner

### ISSUE THREAD

**Issue by ojio-san** [tektoncd/operator#2442]
# Expected Behavior
The tekton resource pruner should pull tkn image and run successfully (previous issue: https://github.com/tektoncd/operator/issues/2368)

# Actual Behavior
ImagePullBackOff on tekton 0.74.0 (also on 0.73.1) 
kubectl events
```
Events:                                                                                              │
│   Type     Reason     Age                From               Message                                  │
│   ----     ------     ----               ----               -------                                  │
│   Normal   Scheduled  63s                default-scheduler  Successfully assigned tekton-pipelines/t │
│ ekton-resource-pruner-77bgh-manual-5nm-g26rm to k3s-01                                               │
│   Normal   Pulling    19s (x3 over 62s)  kubelet            Pulling image "gcr.io/tekton-releases/do │
│ gfooding/tkn:8d4383ff675cdf42b0006fca9b5b3b0c1d6180ff0013805242c877352e03a359"                       │
│   Warning  Failed     18s (x3 over 62s)  kubelet            Failed to pull image "gcr.io/tekton-rele │
│ ases/dogfooding/tkn:8d4383ff675cdf42b0006fca9b5b3b0c1d6180ff0013805242c877352e03a359": rpc error: co │
│ de = NotFound desc = failed to pull and unpack image "gcr.io/tekton-releases/dogfooding/tkn:8d4383ff │
│ 675cdf42b0006fca9b5b3b0c1d6180ff0013805242c877352e03a359": failed to resolve reference "gcr.io/tekto │
│ n-releases/dogfooding/tkn:8d4383ff675cdf42b0006fca9b5b3b0c1d6180ff0013805242c877352e03a359": gcr.io/ │
│ tekton-releases/dogfooding/tkn:8d4383ff675cdf42b0006fca9b5b3b0c1d6180ff0013805242c877352e03a359: not │
│  found                                                                                               │
│   Warning  Failed     18s (x3 over 62s)  kubelet            Error: ErrImagePull                      │
│   Normal   BackOff    5s (x3 over 62s)   kubelet            Back-off pulling image "gcr.io/tekton-re │
│ leases/dogfooding/tkn:8d4383ff675cdf42b0006fca9b5b3b0c1d6180ff0013805242c877352e03a359"              │
│   Warning  Failed     5s (x3 over 62s)   kubelet            Error: ImagePullBackOff 
```

# Steps to Reproduce the Problem

1. Install `tekton-operator-0.74.0` (or 0.73.1)
2. Enable prunner
```
apiVersion: operator.tekton.dev/v1alpha1
kind: TektonConfig
metadata:
  name: config
spec:
  profile: all
  targetNamespace: tekton-pipelines
  pruner:
    resources:
      - pipelinerun
      - taskrun
    keep: ~
    keep-since: 86400
    schedule: "0,30 * * * *"
```

4. Trigger the pruner and see it fails pulling the image

# Additional Info
- Kubernetes version:
```
Client Version: v1.31.0
Kustomize Version: v5.4.2
Server Version: v1.30.6+k3s1
```

- Tekton Pipeline version:
```
tkn version
Client version: 0.39.0
Chains version: v0.23.0
Pipeline version: v0.65.2
Triggers version: v0.30.0
Dashboard version: v0.52.0
```

```
kubectl get pods -n tekton-pipelines -l app=tekton-pipelines-controller -o=jsonpath='{.items[0].metadata.labels.version}'
v0.65.2
```

**Comment by vdemeester** [comment]
@piyush-garg @PuneetPunamiya I think new version of the images are on ghcr.io but… not sure if the one refered is supposed to be on gcr.io or on ghcr.
@afrittoli did we remove some images from gcr.io ?

**Comment by afrittoli** [comment]
I didn't remove any image from `gcr.io`

**Comment by ojio-san** [comment]
Thank you for your replies.

If this can help, I can also reproduce a completely different machine with docker, out of my k3s cluster :
```
$ docker run --rm -it gcr.io/tekton-releases/dogfooding/tkn:8d4383ff675cdf42b0006fca9b5b3b0c1d6180ff0013805242c877352e03a359
Unable to find image 'gcr.io/tekton-releases/dogfooding/tkn:8d4383ff675cdf42b0006fca9b5b3b0c1d6180ff0013805242c877352e03a359' locally
docker: Error response from daemon: manifest for gcr.io/tekton-releases/dogfooding/tkn:8d4383ff675cdf42b0006fca9b5b3b0c1d6180ff0013805242c877352e03a359 not found: manifest unknown: Failed to fetch "8d4383ff675cdf42b0006fca9b5b3b0c1d6180ff0013805242c877352e03a359" from request "/v2/tekton-releases/dogfooding/tkn/manifests/8d4383ff675cdf42b0006fca9b5b3b0c1d6180ff0013805242c877352e03a359".
See 'docker run --help'.
```

Also, I noticed that the latest tag is working:
```
$ docker run --rm -it gcr.io/tekton-releases/dogfooding/tkn:latest sh
Unable to find image 'gcr.io/tekton-releases/dogfooding/tkn:latest' locally
latest: Pulling from tekton-releases/dogfooding/tkn
43c4264eed91: Pull complete
cbfa962fb060: Pull complete
Digest: sha256:d1da68e766393c4b4eb162128f2c5fd2cee270828811a113fcda1e8a586e7471
Status: Downloaded newer image for gcr.io/tekton-releases/dogfooding/tkn:latest
/ #
```

- Could someone reproduce this issue at least with docker ? Or with kubectl run tkn --image=gcr.io/... ?
- Is there a web interface or something, to view all the tags for the gcr.io/tekton-releases/dogfooding/tkn ?

Thank you :)

**Comment by ojio-san** [comment]
Okay, so here is the thing : due to the error, I was thinking that 8d4383ff675cdf42b0006fca9b5b3b0c1d6180ff0013805242c877352e03a359 was the tag, but it was in fact the sha256 for tkn.

So, we can pull this image using : 
```
kubectl run tkn --image=gcr.io/tekton-rel
eases/dogfooding/tkn@sha256:8d4383ff675cdf42b0006fca9b5b3b0c1d6180ff0013805242c877352e03a359 sleep 60
pod/tkn created
```

But, we can see that tekton-resource-pruner is **not** using the `@sha256` prefix, leading to the pull error.
So, @afrittoli was right : no images were removed.


Proofs using a screenshot :
- see the logs on the top of the following capture : tekton-resource-pruner is failing due to the missing `@sha256`
- see the logs on the bottom of the capture : we can actually fetch the image
![image](https://github.com/user-attachments/assets/20ef6644-8413-42ba-b98a-a693cdaf432c)


@vdemeester : I'm not an expert, but how we deal with this issue, now we understand what is the real issue here ?
Is the fix should be applied in one of thoses files https://github.com/tektoncd/operator/pull/2370/files ? or maybe somewhere else ?

Thank you :)

**Comment by vdemeester** [comment]
@ojio-san indeed, it seems https://github.com/tektoncd/operator/pull/2370/files#diff-97c04f73b8960b62c2d0983e533dac7d602066c18bf0cbecacedde337bef6eaaL59 might have broken it (the tag refers to a digest and it doesn't work)

cc @jkandasa @savitaashture @jkhelil

**Comment by jkandasa** [comment]
@vdemeester Someone reported this on 0.73.0 release as well. I fixed it on `0.73.1` seems I forget to fix it on the main branch. I will create PR for that, To fix this in 0.74.x, we have to do a patch release of `0.74.1`
cc @jkhelil
